### PR TITLE
Alerting: Fix case when Alertmanager notifier fails if a URL is not working

### PR DESCRIPTION
**Pull request by kurokochin**
**What this PR does / why we need it**: Fixes behaviour of Notify that returns error when one of the dispatch event return error, to maintain high availability, we should return error when all dispatched events return error instead.

**Which issue(s) this PR fixes**: #30509 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by marefr** [comment]
Please sign the CLA and we'll review this. Thanks

**Comment by kurokochin** [comment]
Hi, sorry, I've signed the CLA, previously my commit not attached to the account that I use to signed the CLA, should be good for now :)

File: pkg/services/alerting/notifiers/alertmanager.go [hunk 1]
```diff
@@ -182,10 +184,16 @@ func (am *AlertmanagerNotifier) Notify(evalContext *alerting.EvalContext) error
...
 			am.log.Error("Failed to send alertmanager", "error", err, "alertmanager", am.Name, "url", url)
-			return err
+			errCnt++
 		}
 	}
 
+	// This happens when every dispatch return error
+	if errCnt == len(am.URL) {
+		am.log.Error("Failed to send alert for all urls")
+		return fmt.Errorf("failed to send alert for all urls")
```
Comment (marefr) [inline pkg/services/alerting/notifiers/alertmanager.go:194]: Think we can skip logging the error in here and just return it

```suggestion
		return fmt.Errorf("failed to send alert to alertmanager")
```

**Review by marefr — CHANGES_REQUESTED** [review]
In general looks good. Added a suggestion.

File: pkg/services/alerting/notifiers/alertmanager.go (hunk 1 above)
Comment (kurokochin) [reply pkg/services/alerting/notifiers/alertmanager.go]: Thanks for your suggestion! Should be resolved :)

**Review by marefr — APPROVED** [review]
LGTM

**Comment by marefr** [comment]
Thank you for contributing to Grafana!

**Comment by kurokochin** [comment]
Glad to help!

**Pull request merged**